cmd/client-fs.go

```diff
@@ -393,6 +399,46 @@ func (f *fsClient) Put(ctx context.Context, reader io.Reader, size int64, progre
 	return f.put(ctx, reader, size, progress, opts)
 }
 
+// checkPathLength - returns error if given path name length more than 255
+func checkPathLength(pathName string) *probe.Error {
+	// Apple OS X path length is limited to 1016
+	if runtime.GOOS == "darwin" && len(pathName) > 1016 {
+		return probe.NewError(errors.New("file name too long"))
+	}
+
+	// Disallow more than 1024 characters on windows, there
+	// are no known name_max limits on Windows.
```

Comment:
UNC paths (used internally by Go) have a total max length of about 32,767 characters.

The details... https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry

In short..

> This allows for paths composed of components separated by backslashes, each up to 255 characters, providing a total path length of up to 32,767 characters.

The sub-paths seems to be checked below.